sale_multi_approval/models/models.py

```diff
@@ -0,0 +1,285 @@
+    relay_state = fields.Binary(string='Relay State', readonly=1)
+    
+    
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    approval_ids = fields.One2many('approval.line', 'sale_order_id')
+    document_fully_approved = fields.Boolean(compute='_compute_document_fully_approved')
+    check_approve_ability = fields.Boolean(compute='_compute_check_approve_ability')
+    is_approved = fields.Boolean(compute='_compute_is_approved')
+    page_visibility = fields.Boolean(compute='_compute_page_visibility')
+    quotation_locked = fields.Boolean()
+
+
+
+    def generate_sale_pdf(self):
+        report_name = self.name
+        report = self.env.ref("sale.action_report_saleorder")
+        report_service = report.report_name
+
+        if report.report_type in ['qweb-html', 'qweb-pdf']:
+            result, format = report._render_qweb_pdf([self.id])
+        else:
+            res = report._render([self.id])
+            result, format = res
+
+        # TODO in trunk, change return format to binary to match message_post expected format
+        result = base64.b64encode(result)
+        ext = "." + format
+        if not report_name.endswith(ext):
+            report_name += ext
+        attachment = (report_name, result)
+        data_attach = {
+            'name': attachment[0],
+            'datas': attachment[1],
+            'res_model': 'sale.order',
+            'res_id': self.id,
+            'type': 'binary',  # override default_type from context, possibly meant for another model!
+        }
+        self.env["ir.attachment"].create(data_attach)
+
+
+    
+
+    @api.depends('approval_ids')
+    def _compute_page_visibility(self):
+        """Compute function for making the approval page visible/invisible"""
+        if self.approval_ids:
+            self.page_visibility = True
+        else:
+            self.page_visibility = False
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        """This is the onchange function of the partner which loads the
+        persons for the approval to the approver table of the account.move"""
+        sale_approval_id = self.env['sale.approval'].search([])
+        self.approval_ids = None
+        if sale_approval_id.approve_customer_sale:
+            vals = {
+                'approver_id': self.user_id.id,
+                'sale_order_id': self.id
+            }
+            line = self.env["approval.line"].sudo().create(vals)
+            self.write({'approval_ids': [(4, line, 0)]})
+
+    @api.depends('approval_ids.approver_id')
+    def _compute_check_approve_ability(self):
+        """This is the compute function which check the current
+        logged in user is eligible or not for approving the document"""
+        current_user = self.env.uid
+        approvers_list = []
+        for approver in self.approval_ids:
+            approvers_list.append(approver.approver_id.id)
+        if current_user in approvers_list:
```

Comment:
Can be a one liner?
self.check_approve_ability = current_user in approvers_list